adapter/dubbo/consumer_filter.go

```diff
@@ -0,0 +1,56 @@
+package dubbo
+
+import (
+	"context"
+	"github.com/apache/dubbo-go/protocol"
+)
+import (
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+)
+
+type consumerFilter struct{}
+
+func (d *consumerFilter) Invoke(ctx context.Context, invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	methodResourceName := getResourceName(invoker, invocation, getConsumerPrefix())
+	interfaceResourceName := ""
+	if getInterfaceGroupAndVersionEnabled() {
+		interfaceResourceName = getColonSeparatedKey(invoker.GetUrl())
+	} else {
+		interfaceResourceName = invoker.GetUrl().Service()
+	}
+	var (
+		interfaceEntry *base.SentinelEntry
+		methodEntry    *base.SentinelEntry
+		b              *base.BlockError
+	)
+
+	if !isAsync(invocation) {
+		interfaceEntry, b = sentinel.Entry(interfaceResourceName, sentinel.WithResourceType(base.ResTypeRPC), sentinel.WithTrafficType(base.Outbound))
+		if b != nil { // blocked
+			return consumerDubboFallback(ctx, invoker, invocation, b)
+		}
+		methodEntry, b = sentinel.Entry(methodResourceName, sentinel.WithResourceType(base.ResTypeRPC), sentinel.WithTrafficType(base.Outbound), sentinel.WithArgs(invocation.Attachments()))
+		if b != nil { // blocked
+			return consumerDubboFallback(ctx, invoker, invocation, b)
+		}
+	} else {
+		// TODO : Need to implement asynchronous current limiting
```

Comment:
What're the differences between async and sync mode?

Reply:
This is mainly to see that in Java code, async entry is used instead of entry to restrict current. At this stage, Dubbo go has not added async field in attachment, so it will not come to async logic

Reply:
Okay. Sentinel Go has unified `entry` and `asyncEntry`, so we could use `sentinel.Entry(args)` for both sync and async scenarios.